lib/cuckoo/core/startup.py

```diff
@@ -92,20 +92,40 @@ def check_webgui_mongo():
         client = connect_to_mongo()
         if not client:
             sys.exit(
-                "You have enabled webgui but mongo isn't working, see mongodb manual for correct installation and configuration\nrun `systemctl status mongodb` for more info"
+                "You have enabled webgui but mongo isn't working, see mongodb "
+                "manual for correct installation and configuration\n"
+                "run `systemctl status mongodb` for more info"
+            )
+
+        # Create separate index for certain fields to enable efficient keyword
+        # searches for large amounts of data in the 'analysis' collection.
+        # NOTE: Silently ignores the creation if the index already exists.
+        items = [
+            "info.id",
+            "info.parent_sample.sha256",
+            "target.file.sha256",
+            "dropped.sha256",
+            "CAPE.payloads.sha256",
+            "procdump.sha256",
+            "procmemory.sha256",
+            "target.file.extracted_files.sha256",
+            "dropped.extracted_files.sha256",
+            "CAPE.payloads.extracted_files.sha256",
+            "procdump.extracted_files.sha256",
+            "procmemory.extracted_files.sha256",
+            "target.file.file_ref",
+            "dropped.file_ref",
+            "CAPE.payloads.file_ref",
+            "procdump.file_ref",
+            "procmemory.file_ref",
+        ]
+        for item in items:
+            mongo_create_index(
```

Comment:
This will take a _very_ long time for a lot of CAPE installations - terabyte scale is common. Probably better to do this in the utils somewhere and emit a warning if the indices aren't found.

Reply:
I'd suggest as a new module in [utils](https://github.com/kevoreilly/CAPEv2/tree/master/utils), `mongodb_indices` or something appropriately named.

In the startup module you can check the indices are there and emit a warning if they're missing.